### DISCUSSION TITLE
Use a content action button to link to the Comments h2

### DISCUSSION THREAD

**Discussion by rachel-rigdon-sp** [squidfunk/mkdocs-material#7050] [Q&A] [answered]
I am adding a comments system to our [docs](https://documentation.sailpoint.com/cam/help/cloud_access_management/config_idp.html#next-steps) and want to use a [content action button](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#code-actions) that will redirect the user to the Comments h2 at the bottom of the page.

This work is similar to https://github.com/squidfunk/mkdocs-material/discussions/6850#discussion-6284949, but I'm not sure how to adapt it since I'm not linking a repository.

**Comment by squidfunk** [comment, marked as answer]
This can be achieve with some quick customizations ☺️ You could override the [`actions.html`](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/actions.html) partial, which contains the buttons, and add a link to an `h2` that you use in the [`comments.html`](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/comments.html) partial. Maybe something like:

`actions.html`

``` jinja2
<a href="#__comments">...</a>
<!-- ... rest of actions.html -->
```

`comments.html`

``` jinja2
<h2 id="__comments">Comments</h2>
<!-- Comment system integration -->
```

**Reply by squidfunk** [reply]
Note that the underscores are just there so they don't clash with any slug that is created from the headlines that are used in Markdown files, as the default slug function will never prepend underscores. It's just a safety net.

**Reply by rachel-rigdon-sp** [reply]
Thank you! that worked! 

Is there an if statement I can add to the actions.html that would exclude the actions button if `comments: false` is in the page meta tag? 

I was able to hide the actions button using https://github.com/squidfunk/mkdocs-material/discussions/6989, but I'd love if there was some logic to simultaneously exclude comments and the actions button.

**Reply by squidfunk** [reply]
Both can just be encapsulated in `page.meta.comments` conditions, [like in our examples](https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/#__codelineno-1-1:2):

`actions.html`

``` jinja2
{% if page.meta.comments %}
  <a href="#__comments">...</a>
{% endif %}
<!-- ... rest of actions.html -->
```

`comments.html`

``` jinja2
{% if page.meta.comments %}
  <h2 id="__comments">{{ lang.t("meta.comments") }}</h2>
  ...
{% endif %}
```

**Comment by rachel-rigdon-sp** [comment]
@squidfunk So I've been able to implement the ~~copy code~~ content.action.edit button, but I'm using instant navigation and I'm having issues making it so that every time you click the `.md-typeset .md-content__button` element, you're redirected to the `meta.comments` h2 at the bottom. 

I know the solution is to add some javascript to my extra.js, but I'm struggling to figure out how to write it. 😞 

I feel like the solution will be similar to https://github.com/squidfunk/mkdocs-material/discussions/6849#discussioncomment-8655697, but less intense. Thank you in advance for your help!!

**Reply by squidfunk** [reply]
I'm confused – the copy code button should not have anything to do with the comments section. If you can share a minimal reproduction of your customizations that showcases the issue, I or somebody else can have a look ☺️

**Reply by rachel-rigdon-sp** [reply]
Omg, I am so sorry. I meant the `content.action.edit` button I'm using as a link to redirect to the comments H2 at the bottom of the page 🤦🏻‍♀️  

Here's the repro:
[9.5.22+insiders.4.53.8-instant-nav-actions-icon.zip](https://github.com/squidfunk/mkdocs-material/files/15312215/9.5.22%2Binsiders.4.53.8-instant-nav-actions-icon.zip)

Here is the same behavior live: https://documentation.sailpoint.com/cam/help/cloud_access_management/dashboard.html

**Reply by kamilkrzyskow** [reply]
You're adding both the button with `herf="#__comments"` and the comments header with `id="#__comments"` in the overrides.
This means that the information doesn't rely on JavaScript, it should always work. However, from testing your remote website it looks like after selecting the comments button, changing site, and selecting comments button again, nothing happens, but the next click after that does work.

I consider this a bug with instant navigation where it for some reason blocks the default jump to the id 🤔

**Reply by rachel-rigdon-sp** [reply]
Thank you for looking into it! 

Should I file this as a bug report? Or is this something that wouldn't fall under MkDocs-Material?

**Reply by kamilkrzyskow** [reply]
Yes, please file in a bug report, but don't use your example with the custom button, as that's in the customization realm.
The same issue happens if you select a specific heading in the ToC on the right, then change page, change back (all using the navigation on the left), and press that same heading with the same `#id` and it will also not work the first time, but will work the second time you click it.

**Reply by rachel-rigdon-sp** [reply]
https://github.com/squidfunk/mkdocs-material/issues/7183

**Reply by squidfunk** [reply]
Indeed, you stumbled upon a bug, thanks for reporting!
